src/ostorlab/serve_app/types.py

```diff
@@ -341,6 +341,231 @@ class Meta:
         )
 
 
+class OxoAgentArgumentType(graphene_sqlalchemy.SQLAlchemyObjectType):
+    """Graphene object type for a list of agent arguments."""
+
+    value = common.Bytes(required=False)
+
+    class Meta:
+        """Meta class for the agent arguments object type."""
+
+        model = models.AgentArgument
+        only_fields = (
+            "id",
+            "name",
+            "type",
+            "description",
+        )
+
+    def resolve_value(
+        self: models.AgentArgument, info: graphql_base.ResolveInfo
+    ) -> bytes:
+        """Resolve agent argument value query.
+
+        Args:
+            self (models.AgentArgument): The agent argument object.
+            info (graphql_base.ResolveInfo): GraphQL resolve info.
+
+        Returns:
+            common.Bytes: The value of the agent argument.
+        """
+        return self.value
+
+
+class OxoAgentArgumentsType(graphene.ObjectType):
+    """Graphene object type for a list of agent arguments."""
+
+    args = graphene.List(OxoAgentArgumentType, required=True)
+
+
+class OxoAgentType(graphene_sqlalchemy.SQLAlchemyObjectType):
+    """SQLAlchemy object type for an agent."""
+
+    args = graphene.Field(OxoAgentArgumentsType, required=True)
+
+    class Meta:
+        """Meta class for the agent object type."""
+
+        model = models.Agent
+        only_fields = (
+            "id",
+            "key",
+        )
+
+    def resolve_args(
+        self: models.Agent, info: graphql_base.ResolveInfo
+    ) -> OxoAgentArgumentsType:
+        """Resolve agent arguments query.
+
+        Args:
+            self (models.Agent): The agent object.
+            info (graphql_base.ResolveInfo): GraphQL resolve info.
+
+        Returns:
+            AgentArgumentsType: List of agent arguments.
+        """
+        with models.Database() as session:
+            args = session.query(models.AgentArgument).filter(
+                models.AgentArgument.agent_id == self.id
+            )
+            return OxoAgentArgumentsType(args=args)
+
+
+class OxoAgentsType(graphene.ObjectType):
+    """Graphene object type for a list of agents."""
+
+    agents = graphene.List(OxoAgentType, required=True)
+    page_info = graphene.Field(
+        common.PageInfo,
+        required=False,
+    )
+
+
+class OxoAgentGroupType(graphene_sqlalchemy.SQLAlchemyObjectType):
+    """SQLAlchemy object type for an agent group."""
+
+    key = graphene.String()
+    agents = graphene.Field(
+        OxoAgentsType,
+        required=True,
+        page=graphene.Int(required=False),
+        number_elements=graphene.Int(required=False),
+    )
+    asset_types = graphene.List(graphene.String)
+
+    class Meta:
+        """Meta class for the agent group object type."""
+
+        model = models.AgentGroup
+
+        only_fields = (
+            "id",
```

Comment:
id?

Reply:
the id is removed from agent_arg and arg, but for agent group i think we should keep it, we might for instance run a scan with the same agent group.